avocado/utils/vmimage.py

```diff
@@ -0,0 +1,340 @@
+        for key in string.Formatter().parse(image_url):
+            if key[1] is not None:
+                required_args.append(key[1])
+        all_args = True
+        for required_arg in required_args:
+            if (required_arg not in self.initargs or
+                    self.initargs[required_arg] is None):
+                all_args = False
+        # In that case, URL is ready
+        if all_args:
+            return image_url.format(**self.initargs)
+
+        # Otherwise, we have to probe it externally
+        # Getting the requested version, if available
+        version = self.get_version()
+        # Getting the image URL
+        return self.get_image(version)
+
+    def get_version(self):
+        pattern = '^%s/$' % self.args['version']
+
+        parser = VMImageHtmlParser(pattern)
+        try:
+            parser.feed(urllib2.urlopen(self.url_versions).read())
+        except:
+            raise ImageProviderError('Cannot open %s' % self.url_versions)
+        if parser.versions:
+            resulting_versions = []
+            for version in parser.versions:
+                try:
+                    resulting_versions.append(int(version))
+                except ValueError:
+                    try:
+                        resulting_versions.append(float(version))
+                    except ValueError:
+                        resulting_versions.append(version)
+            return max(resulting_versions)
+        else:
+            raise ImageProviderError('Version %s not available at %s' %
+                                     (self.default_version, self.url_versions))
+
+    def get_image(self, version):
+        url_images = self.url_images.format(version=version,
+                                            build=self.args['build'],
+                                            arch=self.args['arch'])
+        image = self.image.format(version=version,
+                                  build=self.args['build'],
+                                  arch=self.args['arch'])
+        parser = VMImageHtmlParser(image)
+        try:
+            content = urllib2.urlopen(url_images).read()
+            parser.feed(content)
+        except:
+            raise ImageProviderError('Cannot open %s' % url_images)
+        if parser.versions:
+            return url_images + max(parser.versions)
+        else:
+            raise ImageProviderError("No images matching '%s' at '%s'" %
+                                     (image, url_images))
+
+    def remove(self):
+        """
+        Removes the VM Image
+        """
+        if self.path is not None:
+            os.remove(self.path)
+            self.path = None
+
+    @staticmethod
+    def _create_backed_image(source_image):
+        """
+        Creates a Qcow2 backed image using the original image
+        as backing file.
+        """
+        qemu_img = utils_path.find_command('qemu-img')
+        new_image = '%s-%s' % (source_image, uuid.uuid4())
```

Comment:
please keep the (qcow2) extension at the end.

Reply:
sure